src/api/middleware/utils.js

```diff
@@ -0,0 +1,56 @@
+const HTTPStatus = require("http-status-codes");
+const { APIError, ErrorTypes } = require("./errorHandler");
+const ValidationReasons = require("./validators/validationReasons");
+const lodash = require("lodash");
+
+const DEFAULT_ERROR_CODE = ErrorTypes.VALIDATION_ERROR;
+const DEFAULT_ERROR_MSG = ValidationReasons.UNKNOWN;
+const DEFAULT_STATUS_CODE = HTTPStatus.BAD_REQUEST;
+
+/**
+ * Comibnes array of middleware using OR logic. Only fails if ALL functions fail (either by throwing or calling next(error))
+ *
+ * Each middleware will receive a different req object, no not rely on it to be shared among them
+ *
+ * @param {Function[]} Array of express middleware to be run
+ * @param {object} Options:
+ *  - error_code: error code in case of error (default: ErrorTypes.VALIDATION_ERROR)
+ *  - msg: the message in case of error (default: ValidationReasons.UNKNOWN)
+ *  - status_code: The status used in the HTTP Response in case of error (default: BAD_REQUEST (400))
+ */
+const or = (
+    [...middlewares],
+    {
+        error_code = DEFAULT_ERROR_CODE,
+        msg = DEFAULT_ERROR_MSG,
+        status_code = DEFAULT_STATUS_CODE
+    } = {}
+) => async (initialReq, res, next) => {
+    let success = false;
+    const errors = [];
+    for (const middleware of middlewares) {
+        const req = lodash.cloneDeep(initialReq);
```

Comment:
Maybe we should use this lib in other places that use `JSON.parse(stringify)`, since this seems to be more efficient

Reply:
They don't have the same behavior, since JSON.parse(JSON.stringify(obj)) will not maintain some types such as Dates, or custom class instances or event functions. So it might actually be harder than simply replacing to execute that refactor. Since that is not on the scope of this PR, I won't do it here.

You can create a separate issue to propose that

Reply:
Yes, I was just pointing out something I noticed